components/locid/src/extensions/transform/mod.rs

```diff
@@ -79,6 +79,7 @@ use litemap::LiteMap;
 /// [`Unicode Locale Identifier`]: https://unicode.org/reports/tr35/#Unicode_locale_identifier
 #[derive(Clone, PartialEq, Eq, Debug, Default, Hash, PartialOrd, Ord)]
 #[allow(missing_docs)] // TODO(#1028) - Add missing docs.
+#[allow(clippy::exhaustive_structs)] // this type is stable
```

Comment:
Thought: People shouldn't generally make one of these themselves, so `non_exhaustive` is safe, but I'm okay with it being exhaustive because it is backed by a spec that is unlikely to change in a breaking way.